File: app/src/main/java/com/coinninja/coinkeeper/service/SyncDropBitService.java
package com.coinninja.coinkeeper.service;

import android.app.IntentService;
import android.content.Intent;
import android.util.Log;

import com.coinninja.coinkeeper.cn.account.AccountManager;
import com.coinninja.coinkeeper.service.runner.FulfillSentInvitesRunner;
import com.coinninja.coinkeeper.service.runner.ReceivedInvitesStatusRunner;
import com.coinninja.coinkeeper.service.runner.SyncIncomingInvitesRunner;

import javax.inject.Inject;

import androidx.annotation.Nullable;
import dagger.android.AndroidInjection;

public class SyncDropBitService extends IntentService {
    public SyncDropBitService() {
        this(SyncDropBitService.class.getName());
    }

    public SyncDropBitService(String name) {
        super(name);
    }

    @Inject
    AccountManager accountManager;
    @Inject
    SyncIncomingInvitesRunner syncIncomingInvitesRunner;
    @Inject
    ReceivedInvitesStatusRunner receivedInvitesStatusRunner;
    @Inject
    FulfillSentInvitesRunner fulfillSentInvitesRunner;

    @Override
    public void onCreate() {
        AndroidInjection.inject(this);
        super.onCreate();
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        accountManager.cacheAddresses();
        Log.d("SYNC", "Incoming invites -- Invites Sent to me");
        syncIncomingInvitesRunner.run();
        Log.d("SYNC", "Fulfill invites -- Invites I sent");
        fulfillSentInvitesRunner.run();
        Log.d("SYNC", "Get Fulfilled invites -- Invites Sent to me");
        receivedInvitesStatusRunner.run();
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/service/SyncDropBitService.kt
package com.coinninja.coinkeeper.service

import android.app.IntentService
import android.content.Intent
import android.util.Log
import com.coinninja.coinkeeper.cn.account.AccountManager
import com.coinninja.coinkeeper.model.helpers.WalletHelper
import com.coinninja.coinkeeper.service.runner.FulfillSentInvitesRunner
import com.coinninja.coinkeeper.service.runner.ReceivedInvitesStatusRunner
import com.coinninja.coinkeeper.service.runner.SyncIncomingInvitesRunner
import dagger.android.AndroidInjection
import javax.inject.Inject

class SyncDropBitService @JvmOverloads constructor(name: String = SyncDropBitService::class.java.name) : IntentService(name) {
    @Inject
    lateinit var accountManager: AccountManager
    @Inject
    lateinit var walletHelper: WalletHelper
    @Inject
    lateinit var syncIncomingInvitesRunner: SyncIncomingInvitesRunner
    @Inject
    lateinit var receivedInvitesStatusRunner: ReceivedInvitesStatusRunner
    @Inject
    lateinit var fulfillSentInvitesRunner: FulfillSentInvitesRunner

    override fun onCreate() {
        AndroidInjection.inject(this)
        super.onCreate()
    }

    public override fun onHandleIntent(intent: Intent?) {
        accountManager.cacheAddresses(walletHelper.primaryWallet)
        Log.d("SYNC", "Incoming invites -- Invites Sent to me")
        syncIncomingInvitesRunner.run()
        Log.d("SYNC", "Fulfill invites -- Invites I sent")
        fulfillSentInvitesRunner.run()
        Log.d("SYNC", "Get Fulfilled invites -- Invites Sent to me")
        receivedInvitesStatusRunner.run()
    }
}